### PR TITLE
feat: pass RTL flag to rules function

### PR DESCRIPTION
**Pull request by kuzhelov**
This change is necessary for the sake of component styles being able to adapt to current presence of RTL mode. 

As an example, suppose that we have a Chat Message component with the following layout

image

Relevant thing for us is the fact that 'timestamp' region has left padding, to ensure it won't overlap content of the 'author' area. However, in the RTL mode we would like this padding being applied on the right side of the component, not still on the left one:

image

But, unfortunately, at the moment we are unable to detect whether RTL is enabled from the 'rules' function. What would solve the problem is that if we would pass `isRtl` flag to rules functions, such that the following will be possible:

```js
export default rules = () => {
  root: ({ isRtl }) => {
      padding: isRtl ? .. : ...
  }
}
```

### CODE REVIEW

File: src/components/Provider/Provider.tsx [hunk 1]
```diff
@@ -94,7 +94,7 @@ class Provider extends Component<any, any> {
 
     // ensure we don't assign `undefined` values to the theme context
     // they will override values down stream
-    const theme: any = {}
+    const theme: any = { isRtl: !!this.props.rtl }
```
Comment (levithomason) [inline src/components/Provider/Provider.tsx]: Please destructure `rtl` at the top of the function with the other props.

I propose we don't rename props as they pass from the Provider to the theme.  It adds unnecessary thinking to follow renames.

**Review by levithomason — APPROVED** [review]
Consider comments and merge.

**Comment by levithomason** [comment]
Yep, this is required.  Please see comment and then merge.

**Comment by kuzhelov** [comment]
comments are addressed

**Pull request merged**